init.lua

```diff
@@ -1736,12 +1735,25 @@ function lsp.request_symbol_rename(doc, line, col, new_name)
         callback = function(server, response)
           if response.result and #response.result.changes then
             for file_uri, changes in pairs(response.result.changes) do
-              core.log(file_uri .. " " .. #changes)
-              -- TODO: Finish implement textDocument/rename
+              file_uri = string.sub(file_uri, 8)
+              local buff = core.open_doc(file_uri)
+              core.root_view:open_doc(buff)
+              for i, change in ipairs(changes) do
+                local l1, c1 = change.range.start.line + 1, change.range.start.character + 1
+                local l2, c2 = change.range['end'].line + 1, change.range['end'].character + 1
+                if #buff.selections > 1 then
+                  buff:add_selection(l1, c1, l2, c2)
+                else
+                  buff:set_selection(l1, c1, l2, c2)
+                end
+              end
+              for i, change in ipairs(changes) do
+                local l1, c1 = change.range.start.line + 1, change.range.start.character + 1
+                local l2, c2 = change.range['end'].line + 1, change.range['end'].character + 1
+                buff:replace_cursor(i, l1, c1, l2, c2, function(old) return new_name, true end)
+              end
```

Comment:
Instead of doing this manually, try applying the `TextEdit`s in *reverse order* with `apply_edit`.
See https://github.com/lite-xl/lite-xl-lsp/blob/21faf00bdcb59e5b81ae8b1018d262cbf84efc81/init.lua#L1903-L1910

Reply:
Done